src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+    |is a third breakpoint that is close to both.
+    |
+    |The output file is a tab-delimited table with one record per aggregated cluster of pileups. Aggregated
+    |pileups are reported with the minimum and maximum (inclusive) coordinates of all pileups in the cluster, a
+    |possible putative structural variant event type supported by the pileups, and the sum of read support from all
+    |pileups in the cluster.
+    |""")
+class AggregateSvPileup
+(@arg(flag='i', doc="The input text file of pileups generated by SvPileup") input: FilePath,
+ @arg(flag='o', doc="The output file") output: FilePath,
+ @arg(flag='d', doc="Distance threshold below which to cluster breakpoints") maxDist: Int = 100,
+) extends SvTool with AggregateSvPileupDef {
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    // Read pileups from input file
+    val pileups: Seq[BreakpointPileup] = Metric.read[BreakpointPileup](input)
+
+    // Group pileups by left and right contig and strand
+    val pileupsByContigStrand: Map[BreakpointContigOrientation, PileupGroup] =
+      pileups.groupMap(BreakpointContigOrientation(_))(identity)
+
+    // Within each contig+strand group, identify all immediate "neighbors" of each pileup (other pileups within the
+    // distance threshold)
+    val pileupToNeighbors: Map[BreakpointContigOrientation, Map[BreakpointPileup, PileupGroup]] =
+      pileupsByContigStrand.view.mapValues(AggregateSvPileup.pileupToNeighbors(_, maxDist)).toMap
+
+    // Within each contig+strand group, recursively build (transitive) clusters of neighboring pileups
+    val pileupClusters: Map[BreakpointContigOrientation, Seq[PileupGroup]] =
+      pileupToNeighbors.view.mapValues(AggregateSvPileup.toClusters).toMap
+
+    // Aggregate each cluster
+    val aggregatedPileups: Map[BreakpointContigOrientation, Seq[AggregatedBreakpointPileup]] =
+      pileupClusters.view.mapValues(_.map(AggregatedBreakpointPileup.apply)).toMap
+
+    // Write output file
+    Metric.write[AggregatedBreakpointPileup](output, aggregatedPileups.values.flatten)
+  }
+
+}
+
+
+object AggregateSvPileup extends AggregateSvPileupDef {
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, PileupGroup] = mutable.Map.from(pileups.map((_, Seq())))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.hashCode() < pileup2.hashCode() && isCluster(pileup1, pileup2, maxDist)) {
```

Comment:
A couple of thoughts:
1. I think it might be nice to use `pileup1.total >= pilup2.total` here, so that you tend to put things into groups using the _most_ supported breakpoints as the keys
2. Either way, I think you need to use `<=` or >=` here since two pileups could in theory have a hash collision, in which case you wouldn't end up putting then into the map either way
3. To reduce iteration you could a) turn the PileupGroup into an `IndexedSeq` on load/ splitting by chrom/strand, and then to `for (i <- Range(0, pileups.size); j <- Range(i, pileups.size)) {`.  You could also do this using `pileups.tails` but that is marginally more complicated in my view.

Reply:
I think for (3), the `BreakpointPileup` in `PileupGroup` already have the same `contig` and `strand`, based on how it is being used in the `execute` method (groupby `BreakpointContigOrientation`).  I think this should be stated in the method docs.